### PR TITLE
refactor: respond to PR comment by 0xbok

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: contracts/modules/validators/K1Validator.sol [hunk 1]
```diff
@@ -214,6 +198,24 @@ contract K1Validator is IValidator, ERC7739Validator {
...
+    function _recoverSigner(bytes32 hash, bytes calldata signature) internal view returns (address) {
+        return hash.tryRecover(signature);
+    }
+
+    /// @notice Recovers the signer from an Ethereum signed message
+    /// @param hash The hash of the data to validate
+    /// @param signature The signature data
+    /// @return The recovered signer address
+    /// @notice tryRecover returns address(0) on invalid signature
+    function _recoverSignerFromEthSignedMessage(bytes32 hash, bytes calldata signature) internal view returns (address) {
```
Comment (filmakarov) [inline contracts/modules/validators/K1Validator.sol:215]: we do not need this function

use `_recoverSigner(hash.toEthSignedMessageHash(), signature)` instead

**Review by filmakarov — CHANGES_REQUESTED** [review]

File: contracts/modules/validators/K1Validator.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/modules/validators/K1Validator.sol:215]: ok

Comment (livingrockrises) [reply contracts/modules/validators/K1Validator.sol:215]: makes sense. done

**Review by filmakarov — APPROVED** [review]

File: contracts/modules/validators/K1Validator.sol [hunk 2]
```diff
@@ -252,22 +245,9 @@ contract K1Validator is IValidator, ERC7739Validator {
...
-
-        try this.recoverSignerFromEthSignedMessage(hash, signature) returns (address recoveredSigner) {
-            if (recoveredSigner == owner) {
-                return true;
-            }
-        } catch {
-            // If recovery fails, we'll return false
-        }
-
+        // owner can not be zero address in this contract
```
Comment (0xbok) [inline contracts/modules/validators/K1Validator.sol:248]: can't the owner be zero here: https://github.com/bcnmy/nexus/pull/198/files#diff-0de611a725083823f818063145fae9df78b92f67f62ed9528d5337dd17aa7e10R74-R76 ?

Comment (livingrockrises) [reply contracts/modules/validators/K1Validator.sol:248]: oh, I thought we had zero address check in place. I think we can enforce this in onInstall and create informational issue. @0xbok

Comment (livingrockrises) [reply contracts/modules/validators/K1Validator.sol:248]: https://github.com/bcnmy/nexus/pull/199

**Pull request merged**